File: LeetCode/StockPrice/stockPrice.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        var maxGain = 0;
        var minPriceSoFar = prices[0];

        for(int i = 0; i<prices.Length; i++){
            maxGain = maxGain < (prices[i] - minPriceSoFar) ? (prices[i] - minPriceSoFar) : maxGain;
            minPriceSoFar = minPriceSoFar > prices[i] ? prices[i] : minPriceSoFar;
        }

        return maxGain;
    }
}

File: LeetCode/StockPrice/stockPrice.ts
function maxProfit(prices: number[]): number {
  let maxGain = 0;
  let minPrice = prices[0];
  let n = prices.length;

  for (let i = 0; i < n; i++) {
    maxGain = maxGain < prices[i] - minPrice ? prices[i] - minPrice : maxGain;
    minPrice = minPrice > prices[i] ? prices[i] : minPrice;
  }

  return maxGain;
}
